R/utils.R

```diff
@@ -12,8 +12,34 @@
 #' subdirectory and file(s) within the package.
 #' @note This is an internal function and not intended to be called directly.
 read_csv_file <- function(...) {
-  read.csv(
-    system.file("extdata", ..., package = "fortedata", mustWork = TRUE),
-    stringsAsFactors = FALSE
+  weak_as_tibble(
+    read.csv(
+      system.file("extdata", ..., package = "fortedata", mustWork = TRUE),
+      stringsAsFactors = FALSE
+    )
   )
 }
+
+
+# weak_tibble - use tibble() if available but fall back to
+# data.frame() if necessary
+# not a user-facing function; document via roxygen?
+weak_tibble <- function(..., .force_df = FALSE) {
+  no_tibble <- !suppressWarnings(requireNamespace("tibble", quietly = TRUE))
+  if (.force_df || no_tibble) {
+    data.frame(..., stringsAsFactors = FALSE)
+  } else {
+    tibble::tibble(...)
+  }
```

Comment:
One thing to be careful of here is tat `data.frame` and `tibble::tibble` have different recycling behavior. Namely...

```r
# This works
data.frame(a = 1, b = 1:3, c = 1:9)
# This throws an error
tibble::tibble(a = 1, b = 1:3, c = 1:9)
```

This is an internal helper so it probably doesn't matter, but a slightly safer implementation of this would be something like:

```r
out <- data.frame(..., stringsAsFactors = FALSE)
if (!(.force_df || no_tibble)) out <- weak_as_tibble(out)
out
```

Reply:
Another thing to be careful of is that `tibble::tibble` lets you use previous arguments in subsequent ones (similar to `dplyr::mutate`), but `data.frame` does not:

```r
# this works
tibble::tibble(x = 1:3, y = x^2)
# this does not
data.frame(x = 1:3, y = x^2)
```